FileShareAdminClient/FileShareApiAdminClient.cs

```diff
@@ -17,21 +19,23 @@ namespace UKHO.FileShareAdminClient
 {
     public interface IFileShareApiAdminClient : IFileShareApiClient
     {
-        Task<HttpResponseMessage> AppendAclAsync(string batchId, Acl acl, CancellationToken cancellationToken);
-        Task<IBatchHandle> CreateBatchAsync(BatchModel batchModel, CancellationToken cancellationToken);
+        Task<IBatchHandle> CreateBatchAsync(BatchModel batchModel, CancellationToken? cancellationToken = null);
         Task<BatchStatusResponse> GetBatchStatusAsync(IBatchHandle batchHandle);
-
-        Task AddFileToBatch(IBatchHandle batchHandle, Stream stream, string fileName, string mimeType, CancellationToken cancellationToken,
+        Task AddFileToBatch(IBatchHandle batchHandle, Stream stream, string fileName, string mimeType,
+            params KeyValuePair<string, string>[] fileAttributes);
+        Task AddFileToBatch(IBatchHandle batchHandle, Stream stream, string fileName, string mimeType, CancellationToken? cancellationToken = null,
             params KeyValuePair<string, string>[] fileAttributes);
-
         Task AddFileToBatch(IBatchHandle batchHandle, Stream stream, string fileName, string mimeType,
-            Action<(int blocksComplete, int totalBlockCount)> progressUpdate, CancellationToken cancellationToken,
+            Action<(int blocksComplete, int totalBlockCount)> progressUpdate,
+            params KeyValuePair<string, string>[] fileAttributes);
+        Task AddFileToBatch(IBatchHandle batchHandle, Stream stream, string fileName, string mimeType,
```

Comment:
Ideally I would like to see all newly added interface methods return `IResult`, but realise it will cause you extra work in the case of the new `AddFileToBatch()` methods.

If possible please refactor the newly added AddFileToBatch methods to return `IResult` if it does not cause you too much additional work

Reply:
Changes related to existing endpoints will be handled in a separate PBI. Hence, AddFileToBatch() method will be covered in that PBI.